pipeline/transformer/cifs.py

```diff
@@ -130,19 +132,62 @@ def _should_skip(self, situation, situationRecord):
 
         return False
 
+    def _laneStatusCoded(self, situationRecord) -> Optional[str]:
+        """
+        <situationRecord id="xxx">
+            <impact>
+                <impactExtension>
+                    <impactExtended>
+                        <laneStatusCoded>o2xx</laneStatusCoded>
+                        <laneRestriction>
+                            <lane>allLanesCompleteCarriageway</lane>
+                        </laneRestriction>
+                    </impactExtended>
+                </impactExtension>
+            </impact>
+        </situationRecord>
+        """
+        lsElement = situationRecord.find('d:impact/d:impactExtension/d:impactExtended/d:laneStatusCoded', ns)
+        return lsElement.text if lsElement is not None else None
+
+    @staticmethod
+    def _is_opposite_direction_concerned(lanestatus: str) -> bool:
+        # lanes can be single carriageways (encoded by centre line '1', or dual carriagways (encoded by '2'))
+        # we split at both.
+        lanesPerDirection = lanestatus.replace('2', '1').split('1')
```

Comment:
Can you add tests for this logic? That would make it a lot easier to grasp.

Reply:
Done with e57911d.

To test time dependant filtering of constructions, I added a new parameter `current_time` so the point in time can be fixed for the test.